src/stan/model/indexing/lvalue.hpp

```diff
@@ -154,7 +153,7 @@ assign(Eigen::Matrix<T, Eigen::Dynamic, 1>& x,
  * the indexed size.
  */
 template <typename T, typename I, typename U>
-inline typename boost::disable_if<boost::is_same<I, index_uni>, void>::type
+inline typename std::enable_if<!std::is_same<I, index_uni>::value, void>::type
```

Comment:
`enable_if_t` returns `void` by default so these can be 

```suggestion
inline std::enable_if_t<!std::is_same<I, index_uni>::value>
```

Reply:
Ah, nice!